pkg/keyspace/tso_keyspace_group.go

```diff
@@ -286,33 +290,70 @@ func (m *GroupManager) GetKeyspaceGroupByID(id uint32) (*endpoint.KeyspaceGroup,
 // DeleteKeyspaceGroupByID deletes the keyspace group by ID.
 func (m *GroupManager) DeleteKeyspaceGroupByID(id uint32) (*endpoint.KeyspaceGroup, error) {
 	var (
-		kg  *endpoint.KeyspaceGroup
-		err error
+		kg           *endpoint.KeyspaceGroup
+		err          error
+		moreToDelete = true
 	)
 
 	m.Lock()
 	defer m.Unlock()
-	if err := m.store.RunInTxn(m.ctx, func(txn kv.Txn) error {
-		kg, err = m.store.LoadKeyspaceGroup(txn, id)
+	for moreToDelete {
+		err = m.store.RunInTxn(m.ctx, func(txn kv.Txn) error {
+			kg, err = m.store.LoadKeyspaceGroup(txn, id)
+			if err != nil {
+				return err
+			}
+			if kg == nil {
+				return nil
+			}
+			if kg.IsSplitting() {
+				return ErrKeyspaceGroupInSplit
+			}
+
+			var (
+				keyspaceIDsToUnlock = make([]uint32, 0, maxTxnOps)
+				deleteKeyspaceCount = 0
+			)
+			defer func() {
+				for _, id := range keyspaceIDsToUnlock {
+					m.km.metaLock.Unlock(id)
+				}
+			}()
+			for _, id := range kg.Keyspaces {
+				if deleteKeyspaceCount >= maxTxnOps {
```

Comment:
How do we count txn ops? For every increment of deleteKeyspaceCount, given the fact that it calls m.km.store.LoadKeyspaceMeta(txn, id) and err = m.km.store.SaveKeyspaceMeta(txn, ks), it's counted two txn ops or just one (only write ops)? If it's counted two txn ops, the transaction size will be double of maxTxnOps